File: zad1.cpp
// регулярные выражения
#include <iostream>
#include <regex>

using namespace std;

int main() {
    
    string str = "1001tf11flb10001"; // строка
    regex regular("1(0+)1"); // условие поиска
    smatch result; // результат

    while (regex_search(str, result, regular)) {
        cout << result[0] << endl; // возвращает первое совпадение с регулярным выражением
        str = result.suffix().str(); // возвращает оставшуюся часть строки после последнего соответствия регулярному выражению
    }

    return 0;
}

File: zad1.rs
use regex::Regex;

fn main() {
    let str = "1001tf11flb10001"; // строка
    let regular = Regex::new(r"1[0]+1").unwrap(); // условие поиска
    
    for mat in regular.find_iter(str) { // вывод найденных подстрок
        println!("{}", mat.as_str()); 
    }
}